x-pack/plugins/apm/server/lib/service_map/get_service_map_from_trace_ids.ts

```diff
@@ -257,22 +257,21 @@ export async function getServiceMapFromTraceIds({
     });
   }
 
-  const connections = uniq(
-    paths.flatMap((path) => {
-      return path.reduce((conns, location, index) => {
-        const prev = path[index - 1];
-        if (prev) {
-          return conns.concat({
-            source: prev,
-            destination: location,
-          });
-        }
-        return conns;
-      }, [] as Connection[]);
-    }, [] as Connection[]),
-    (value, _index, array) => {
-      return find(array, value);
-    }
+  const connectionsArr = paths.flatMap((path) => {
+    return path.reduce((conns, location, index) => {
+      const prev = path[index - 1];
+      if (prev) {
+        return conns.concat({
+          source: prev,
+          destination: location,
+        });
+      }
+      return conns;
+    }, [] as Connection[]);
+  }, [] as Connection[]);
+
+  const connections = uniqBy(connectionsArr, (value) =>
+    find(connectionsArr, value)
   );
```

Comment:
@ogupte Do we have unit tests to safeguard this?

Reply:
In APM, we should probably externalize this logic into its own function so we can add tests for it.

Reply:
@ogupte talked with you offline. I'll be waiting your PR against that branch 😃

Reply:
PR is here: https://github.com/mistic/kibana/pull/7

Reply:
@sqren would like you to review it as well.

Reply:
@ogupte thank you very much for your help! ❤️ I'm available to merge them right now here, should we wait for @sqren review on them?

Reply:
Thanks both. PR looks good ✅